js/ui/codingManager.js

```diff
@@ -0,0 +1,583 @@
+        dst.pivot_point = new Clutter.Point({ x: 0.5, y: 0.5 });
+        src.pivot_point = new Clutter.Point({ x: 0.5, y: 0.5 });
+
+        if (srcGeometry.equal(dstGeometry))
+            return;
+
+        dst.meta_window.move_resize_frame(false,
+                                          srcGeometry.x,
+                                          srcGeometry.y,
+                                          srcGeometry.width,
+                                          srcGeometry.height);
+    },
+
+    _animate: function(src, dst, direction) {
+        // Tween both windows in a rotation animation at the same time
+        // with backface culling enabled on both. This will allow for
+        // a smooth transition.
+        Tweener.addTween(src, {
+            rotation_angle_y: direction == Gtk.DirectionType.RIGHT ? 180 : -180,
+            time: WINDOW_ANIMATION_TIME * 4,
+            transition: 'easeOutQuad',
+            onComplete: Lang.bind(this, this.rotateOutCompleted, src)
+        });
+        Tweener.addTween(dst, {
+            rotation_angle_y: 0,
+            time: WINDOW_ANIMATION_TIME * 4,
+            transition: 'easeOutQuad',
+            onComplete: Lang.bind(this, this.rotateInCompleted, dst)
+        });
+
+        // Gently fade the window in, this will paper over
+        // any artifacts from shadows and the like
+        //
+        // Note: The animation time is reduced here - this
+        // is intention. It is just to prevent initial
+        // "double shadows" when rotating between windows.
+        Tweener.addTween(dst, {
+            opacity: 255,
+            time: WINDOW_ANIMATION_TIME,
+            transition: 'linear'
+        });
+    },
+
+    _removeEffect: function(list, actor) {
+        let idx = list.indexOf(actor);
+        if (idx != -1) {
+            list.splice(idx, 1);
+            return true;
+        }
+        return false;
+    },
+
+    _removeSession: function(session) {
+        let idx = this._sessions.indexOf(session);
+        if (idx != -1) {
+            this._sessions.splice(idx, 1);
+            return true;
+        }
+        return false;
+    },
+
+    rotateInCompleted: function(actor) {
+        if (this._removeEffect(this._rotateInActors, actor)) {
+            Tweener.removeTweens(actor);
+            actor.opacity = 255;
+            actor.rotation_angle_y = 0;
+            actor.set_cull_back_face(false);
+        }
+    },
+
+    rotateOutCompleted: function(actor) {
+        if (this._removeEffect(this._rotateOutActors, actor)) {
+            Tweener.removeTweens(actor);
+            actor.hide();
+            actor.rotation_angle_y = 0;
+            actor.set_cull_back_face(false);
+        }
+    },
+
+    _getAppManifest: function(flatpakID) {
+        function getAppManifestAt(location, flatpakID) {
+            let manifestFile = Gio.File.new_for_path(GLib.build_filenamev([location, '/app/', flatpakID, '/current/active/files/manifest.json']));
```

Comment:
When using `GLib.build_filenamev()` you should not specify path delimiters manually, similar to e.g. `os.path.join()` in python. You can just pass an array with all the path components.